src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -189,12 +187,7 @@ public void start(final Map<String, String> props) {
                             Schema.STRING_SCHEMA, json));
                 });
 
-                // If the cursor is invalidated add the record and return calls
-                if (changeStreamDocument.getString("operationType", new BsonString("")).getValue().equalsIgnoreCase(INVALIDATE)) {
```

Comment:
Can you explain why this technique didn't work? I thought it would.

Reply:
It did work - just requires users to not filter out any operation types.  The work around is to always include the Invalidate event as well.